frontend/app/profile/edit/page.tsx

```diff
@@ -0,0 +1,279 @@
+              break;
+            case 503:
+              message =
+                "The service is currently unavailable. Please try again later.";
+              break;
+            default:
+              message = "An unexpected error occurred. Please try again.";
+          }
+        }
+
+        showErrorToast(title, message);
+      });
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value, files } = e.target as HTMLInputElement;
+    if (name === "profilePicturePath" && files?.[0]) {
+      const file = files[0];
+
+      // Validate the image file
+      const validation = validateImageFile(file);
+      if (!validation.isValid) {
+        showErrorToast("Invalid File", validation.error ?? "Invalid file");
+        return;
+      }
+
+      const imageUrl = URL.createObjectURL(file);
+      setSelectedImage(imageUrl);
+      setPlayerProfileRequest((prev) => ({
+        ...prev,
+        profileUpdates: {
+          ...prev.profileUpdates,
+          profilePicturePath: imageUrl,
+        },
+        profileImage: file,
+      }));
+    } else {
+      setPlayerProfileRequest((prev) => ({
+        ...prev,
+        profileUpdates: {
+          ...prev.profileUpdates,
+          [name]: value || null,
+        },
+      }));
+    }
+  };
+
+  const handleClearImage = () => {
+    setSelectedImage(null);
+    setPlayerProfileRequest((prev) => ({
+      ...prev,
+      profileUpdates: {
+        ...prev.profileUpdates,
+        profilePicturePath:
+          PROFILE_IMAGE_API + playerProfileRequest.profileUpdates.username,
+      },
+      profileImage: new File([], ""),
+    }));
+  };
+
+  if (isLoading) return <div>Loading...</div>;
+
+  return (
+    <div className="flex flex-col items-center min-h-screen mt-24">
+      <Card className="w-full flex flex-col max-w-2xl">
+        <form onSubmit={handleSubmit}>
+          <CardHeader>
+            <CardTitle className="text-center">Edit Profile</CardTitle>
+          </CardHeader>
+
+          <CardContent>
+            {/* Profile Picture */}
+            <ProfilePicture
+              selectedImage={selectedImage}
+              profilePicturePath={
+                playerProfileRequest.profileUpdates.profilePicturePath
+              }
+              handleChange={handleChange}
+              handleClearImage={handleClearImage}
+            />
+
+            {/* First Name */}
+            <div className="mb-4">
+              <label htmlFor="firstName" className="block text-xl mb-2">
+                First Name:
+              </label>
+              <input
+                type="text"
+                id="firstName"
+                name="firstName"
+                value={playerProfileRequest.profileUpdates.firstName || ""}
+                onChange={handleChange}
+                className="w-full p-2 rounded-lg bg-[#333333] text-white"
```

Comment:
shouldnt hardcode bg-[hexcolor], should change to use bg-secondary or bg-primary-foreground or something

Reply:
will be refactored in the next sprint